Mathlib/LinearAlgebra/Matrix/Swap.lean

```diff
@@ -0,0 +1,167 @@
+/-
+Copyright (c) 2024 Judith Ludwig, Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Judith Ludwig, Christian Merten
+-/
+import Mathlib.LinearAlgebra.Matrix.GeneralLinearGroup.Defs
+import Mathlib.Data.Matrix.PEquiv
+
+/-!
+# Swap matrices
+
+A swap matrix indexed by `i` and `j` is the matrix that, when multiplying another matrix
+on the left (resp. on the right), swaps the `i`-th row with the `j`-th row
+(resp. the `i`-th column with the `j`-th column).
+
+Swap matrices are a special case of *elementary matrices*. For transvections see
+`Mathlib.LinearAlgebra.Matrix.Transvection`.
+
+## Implementation detail
+
+This is a thin wrapper around `(Equiv.swap i j).toPEquiv.toMatrix`.
+-/
+
+namespace Equiv
+
+open Matrix
+
+variable {α n m : Type*}
+
+lemma toPEquiv_toMatrix_mulVec_apply [Fintype n] [DecidableEq n]
+    [NonAssocSemiring α] (σ : m ≃ n) (i : m) (a : n → α) :
+    (σ.toPEquiv.toMatrix *ᵥ a) i = a (σ i) := by
```

Comment:
I think this might be slightly more useful as
```lean
lemma toPEquiv_toMatrix_mulVec [Fintype n] [DecidableEq n]
    [NonAssocSemiring α] (σ : m ≃ n) (a : n → α) :
    σ.toPEquiv.toMatrix *ᵥ a = a ∘ σ
```
though it's ok to have both as long as you prove one with the other.